tools/proofers/codepoint_validator.js

```diff
@@ -33,18 +33,22 @@ $(function () {
             charClass += utf8Chr(code[0]) + "-" + utf8Chr(code[1]);
         }
     });
-    var pattern = new RegExp("[^" + charClass + "]", "ug");
+
+    var badPattern = new RegExp("[^" + charClass + "]", "ug");
```

Comment:
Do you need two of these? I would think that doing a global for where you use badPattern1 would also work.

Reply:
I thought it would work but on successive test functions it starts from the lastIndex from the previous test so doesn't find anything. I've now reset the lastIndex before test instead of using a different regex.